src/erlcloud_sqs.erl

```diff
@@ -420,7 +420,7 @@ sqs_request(Config, QueueName, Action, Params) ->
             erlang:error({aws_error, Reason})
     end.
 
-queue_path([$/|_] = QueueName) -> QueueName;
+queue_path([$/|QueueName]) -> [$/ |erlcloud_http:url_encode(QueueName)];
 queue_path([$h,$t,$t,$p|_] = URL) ->
     re:replace(URL, "^https?://[^/]*", "", [{return, list}]);
```

Comment:
Ignore the comment above .Actually correct url should have special chars encoded already